bodhi/server/templates/update.html

```diff
@@ -98,122 +99,121 @@
     var receive_resultsdb = function(data) {
+      if (data.next){
+        request_results(data.next);
+      }
+      activeResultsAjaxCalls--;
 
       // First, prune duplicate results.  For instance, some depcheck runs
       // happen multiple times on an update.  We only (imho) want to display
       // the latest ones for each arch.  So, prune like this:
-      var changed = false;
+      //var changed = false;
       $.each(data.data, function(i, result) {
 
         // However, we want to skip over any ABORTED results:
         // https://github.com/fedora-infra/bodhi/issues/167
         if (result.outcome == 'ABORTED') return;
 
         var name = result.testcase.name;
-        var arch = result.result_data.arch;
-        var item = result.result_data.item;
+        var arch = result.data.arch;
+        var item = result.data.item;
         var submit_time = new Date(result.submit_time);
-        if (latest[name] === undefined)
-          latest[name] = {};
-        if (latest[name][arch] === undefined)
-          latest[name][arch] = {};
-        if (latest[name][arch][item] === undefined) {
-          latest[name][arch][item] = result;
-          changed = true;
-        }
-        if (new Date(latest[name][arch][item].submit_time) < submit_time) {
-          latest[name][arch][item] = result;
-          changed = true;
-        }
-      });
 
-      // If this was the last page of data, then bail out.
-      if (data.data.length == 0) {
-        // Kill the thing showing that we're still loading data, since we're
-        // done-done now.
-        $('#resultsdb .spinner').remove();
-
-        // Furthermore, if no rows got written.. then we have no results at
-        // all!  So, hang an 'out to lunch' sign on the door.
-        if ($('#resultsdb tr:not(.warning)').length == 0) {
-          var testcase = data.href.match(/testcases\/(.*)\/results/)[1];
-          if ($.inArray(testcase, requirements) != -1) {
-            $('#resultsdb table').append(make_row(
-                  'ABSENT',
-                  testcase,
-                  'No result found',
-                  undefined,
-                  undefined,
-                  data.href
-            ));
+
+        if (latest[item] === undefined)
+          latest[item] = {};
+        if (latest[item][arch] === undefined)
+          latest[item][arch] = {};
+        if (latest[item][arch][name] === undefined) {
+          latest[item][arch][name] = result;
+
+          // the latest/ api endpoint doesnt take into account multiple arches
+          // for dist.depcheck. So if we hit one of these for the first time
+          // we do a call to get the other one.
+          if (name == 'dist.depcheck'){
+            var theurl = base_url+"results?testcases=dist.depcheck&item="+item+"&type=koji_build&limit=1";
+            if (arch == 'x86_64'){
+              theurl = theurl +"&arch=i386"
+            } else {
+              theurl = theurl +"&arch=x86_64"
+            }
```

Comment:
@kparal, @rajcze: You mentioned ```x86_64``` and ```i386``` - are there also results for the other arches, like aarch64? I'm wondering if we should avoid hardcoding arches here, or if we should at least look for results for more arches than just these two. What do you think?